zcncore/transaction_query_base.go

```diff
@@ -329,6 +239,51 @@ func (tq *TransactionQuery) FromAny(ctx context.Context, query string) (QueryRes
 
 }
 
+// FromAll query transaction from all sharders whatever it is selected or offline in previous queires, and return consensus result
+func (tq *TransactionQuery) FromAll(ctx context.Context, query string, handle QueryResultHandle) error {
+	if tq == nil || tq.max == 0 {
+		return ErrNoAvailableSharders
+	}
+
+	urls := make([]string, 0, tq.max)
+	for _, host := range tq.sharders {
+		urls = append(urls, tq.buildUrl(host, query))
+	}
+
+	r := resty.New(resty.WithTimeout(10 * time.Second))
```

Comment:
We'd better use the `ctx` instead of coming up with a new timeout, the API caller should be able to control the timeout with the `ctx` param. Anyway, that will require the refactoring of the. `resty` package to drop the timeout param or make it optional, which could be done in another thread.